chrome/content/zotero/xpcom/date.js

```diff
@@ -623,14 +623,12 @@ Zotero.Date = new function(){
 		return string;
 	}
 	
-	this.strToISO = function (str) {
-		var date = this.strToDate(str);
-		
-		if(date.year) {
+	this.dateToISO = function (date) {
```

Comment:
It would be good to avoid adding a `dateTo` function that's not referring to JS `Date` — we have `strToDate` and `formatDate` that refer to our weird custom object, but the other `dateTo*` functions take `Date` objects. Maybe `datePartsToISO`?